samples/wifi/scan/src/main.c

```diff
@@ -221,44 +244,44 @@ static int wifi_scan(void)
 {
 	struct net_if *iface = net_if_get_default();
 
-	for (int i = 0; i < ARRAY_SIZE(tests); i++) {
-		struct wifi_scan_params params = tests[i];
-#ifdef CONFIG_WIFI_SCAN_SSID_FILT_MAX
-		if (!strlen(params.ssids[0])) {
-			char *buf = NULL;
-
-			buf = malloc(strlen(CONFIG_WIFI_SCAN_SSID_FILT) + 1);
-			if (!buf) {
-				LOG_ERR("Malloc Failed");
-				return -EINVAL;
-			}
-
-			strcpy(buf, CONFIG_WIFI_SCAN_SSID_FILT);
-			if (wifi_utils_parse_scan_ssids(buf,
-							params.ssids)) {
-				LOG_ERR("Incorrect value(s) in CONFIG_WIFI_SCAN_SSID_FILT: %s",
-						CONFIG_WIFI_SCAN_SSID_FILT);
-				return -EINVAL;
-			}
+	struct wifi_scan_params params = tests[0];
+
+	if (strlen(CONFIG_WIFI_SCAN_BANDS_LIST)) {
+		char *buf = malloc(strlen(CONFIG_WIFI_SCAN_BANDS_LIST) + 1);
+
+		if (!buf) {
+			LOG_ERR("Malloc Failed");
+			return -EINVAL;
 		}
-#endif
-#ifdef CONFIG_WIFI_SCAN_CHAN
-		if (wifi_utils_parse_scan_chan(CONFIG_WIFI_SCAN_CHAN_LIST,
-				params.chan)) {
-			LOG_ERR("Chan Parse failed");
+		strcpy(buf, CONFIG_WIFI_SCAN_BANDS_LIST);
+		if (wifi_utils_parse_scan_bands(buf, &params.bands)) {
+			LOG_ERR("Incorrect value(s) in CONFIG_WIFI_SCAN_BANDS_LIST: %s",
+					CONFIG_WIFI_SCAN_BANDS_LIST);
+			free(buf);
 			return -ENOEXEC;
 		}
-#endif
-		if (net_mgmt(NET_REQUEST_WIFI_SCAN, iface, &params,
-				sizeof(struct wifi_scan_params))) {
-			LOG_ERR("Scan request failed");
+		free(buf);
+	}
+
+	if (strlen(CONFIG_WIFI_SCAN_CHAN_LIST)) {
```

Comment:
Do not repeat strlen on the same variable/cons if it does not change, cache result of first call.
Also, here is literal used and I guess sizeof could get the size of this at compile time.